pkgs/development/tools/analysis/modex/default.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchurl, bison, flex }:
+
+stdenv.mkDerivation rec {
+  version = "2.10";
+  name = "modex-${version}";
+
+  src = fetchurl {
+    url = "http://spinroot.com/modex/modex${version}.tar.gz";
+    sha256 = "1iz41w926jgx7b42cmfv7v3i4qwdldababqv60f54bisvrq7psws";
+  };
+
+  preConfigure = ''
+    substituteInPlace makefile --replace /usr/local/modex $out/share/modex
+    substituteInPlace makefile --replace /usr/local $out
+    substituteInPlace xtract.c --replace /usr/local $out/share
+    '';
+
+  sourceRoot = "Src2.10";
+
+  preInstall = ''
+    mkdir -p $out/bin
+    mkdir -p $out/share/modex/Examples
+    mkdir -p $out/share/modex/Manual
+    '';
+
+  postInstall = ''
+    install -m755 ../Scripts/verify $out/bin/
+    install -m644 ../Examples/* $out/share/modex/Examples/
+    install -m644 ../Manual/* $out/share/modex/Manual/
+    '';
+  
+  propagatedBuildInputs = [ ];
+
+  buildInputs = [ bison flex ];
+
+  meta = {
+    description = "Mechanical extraction of verification models from implementation level C code for spin.";
+    platforms = stdenv.lib.platforms.all;
+    license = stdenv.lib.licenses.free;
```

Comment:
Looks like this is `unfreeRedistributable`, from modex.c:

``` c
/* Copyright (c) 2000-2003 by Lucent Technologies - Bell Laboratories     */
/* All Rights Reserved.  This software is for educational purposes only.  */
/* Permission is given to distribute this code provided that this intro-  */
/* ductory message is not removed and no monies are exchanged.            */
/* No guarantee is expressed or implied by the distribution of this code. */
/* Software written by Gerard J. Holzmann based on the public domain      */
/* ANSI-C parser Ctree Version 0.14 from Shaun Flisakowski                */
/* Extensions (c) 2004-2014 by Caltech/JPL Pasadena, CA 91109             */
```

The "no monies" part I would say makes it nonfree.